modules/distribution/utils/delegator_rewards.go

```diff
@@ -7,14 +7,26 @@ import (
 	"github.com/desmos-labs/juno/client"
 
 	"github.com/forbole/bdjuno/database"
-	"github.com/forbole/bdjuno/types"
+	"github.com/forbole/bdjuno/types/config"
 
 	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	"github.com/forbole/bdjuno/types"
 	"github.com/rs/zerolog/log"
 )
 
 // UpdateDelegatorsRewardsAmounts updates the delegators commission amounts
-func UpdateDelegatorsRewardsAmounts(height int64, client distrtypes.QueryClient, db *database.Db) {
+func UpdateDelegatorsRewardsAmounts(height int64, client distrtypes.QueryClient, db *database.Db, cfg *config.Config) {
+	rewards, _ := db.GetDelegatorRewards()
+	interval := cfg.GetRewardsFrequencyConfig().GetRewardsFrequency()
+
+	if interval > 0 {
+		if len(rewards) == 0 || height%interval == 0 {
+			go updateDelegatorsRewards(height, client, db)
+		}
+	}
```

Comment:
Instead of putting the longer term inside the `if` statement, it's more efficient to return immediately if the condition is not satified: 

```suggestion
	interval := cfg.GetRewardsFrequencyConfig().GetRewardsFrequency()
	if interval == 0 {
		log.Debug().Str("module", "distribution").Msg("delegator rewards refresh interval set to 0. Skipping refresh")
		return
	}
	
	rewards, _ := db.GetDelegatorRewards()
	if len(rewards) == 0 || height%interval == 0 {
		go updateDelegatorsRewards(height, client, db)
	}
```

Reply:
Updated